python/ray/autoscaler/_private/command_runner.py

```diff
@@ -465,10 +465,8 @@ def _run_helper(
                     silent=silent,
                     use_login_shells=is_using_login_shells(),
                 )
-            if with_output:
-                return self.process_runner.check_output(final_cmd)
             else:
-                return self.process_runner.check_call(final_cmd)
+                return self.process_runner.check_output(final_cmd)
```

Comment:
We're in the branch that says we need the output, yet this used to do check_call.
I'm actually not sure if I'm breaking something by fixing it.

Reply:
I guess I'd better at least try launching a cluster with these changes before merging.

Reply:
I mean, we 100% don't rely on this returning exit code 0.
I wonder how anything ever worked if all of the docker logic requires using "with_output=True".

Reply:
We do have integration tests for the docker command runner logic. And I will go ahead and do a manual test.

Reply:
I'm actually going to change this back and leave a note.